rtil/src/threads/ue/rebo/rebo_init.rs

```diff
@@ -1486,3 +1507,57 @@ fn timer_set_game_time(time: f64) {
 fn timer_pause_game_time() {
     Timer::pause_game_time();
 }
+#[rebo::function("Tas::timer_create_segment")]
+fn timer_create_segment(name: String) {
+    Run::create_segment(name);
+}
+#[rebo::function("Tas::timer_remove_segment")]
+fn timer_remove_segment(index: i32) {
+    Run::remove_segment(index);
+}
+#[rebo::function("Tas::timer_get_game_name")]
+fn timer_get_game_name() -> String {
+    Run::game_name()
+}
+#[rebo::function("Tas::timer_set_game_name")]
+fn timer_set_game_name(game: Games) {
+    Run::set_game_name(game);
+}
+#[rebo::function("Tas::timer_get_category_name")]
+fn timer_get_category_name() -> String {
+    Run::category_name()
+}
+#[rebo::function("Tas::timer_set_category_name")]
+fn timer_set_category_name(category: String) {
+    Run::set_category_name(category);
+}
+#[rebo::function("Tas::timer_get_segments")]
+fn timer_get_segments() -> Vec<Segment> {
+    let mut segments = Vec::new();
+    for mut seg in Run::get_segments() {
+        let mut comp_names = Vec::new();
+        let mut comp_times = Vec::new();
+        for comparison in seg.comparisons_mut().iter() {
+            comp_names.push(comparison.to_owned().0.to_string());
+            comp_times.push(comparison.to_owned().1.game_time.unwrap().total_seconds());
+        }
+        let segment = Segment {
+            name: seg.name().to_string(),
+            time: seg.split_time().game_time.unwrap().total_seconds(),
+            pb_time: seg.personal_best_split_time().game_time.unwrap().total_seconds(),
+            best_time: seg.best_segment_time().game_time.unwrap().total_seconds(),
```

Comment:
I hope all these (and above) unwraps don't hurt us. What if an empty new run / lss file is loaded? I think those unwraps might crash.